be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -47,7 +52,64 @@ case object RollCallHandler extends MessageHandler {
   }
 
   def handleCloseRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
-    val ask: Future[GraphMessage] = dbAskWritePropagate(rpcMessage)
-    Await.result(ask, duration)
+    rpcMessage.getParamsMessage match {
+      case Some(message: Message) =>
+        val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
+        val unknownAnswer: String = "Database actor returned an unknown answer"
+
+        //Creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a GraphMessage
+        def createAttendeeChannels(attendees: List[PublicKey], rpcMessage: JsonRpcRequest): GraphMessage = {
+          attendees match {
+            case Nil => Left(rpcMessage)
+            case head::tail => 
+              //a closeRollCall is always sent in the lao's main channel so we are allowed to do this
+              val socialChannel: String = generateSocialChannel(rpcMessage.getParamsChannel, head)
+              val ask: Future[GraphMessage] = (dbActor ? DbActor.CreateChannel(Channel(socialChannel), ObjectType.CHIRP)).map {
+                case DbActorWriteAck() => createAttendeeChannels(tail, rpcMessage)
+                //FIXME: Once the real PoP tokens are implemented, the same person will not have access to the same channel name twice in a row, so we can remove this line
+                //however, this is to prevent errors for now within a LAO for multiple Roll Calls and the same participants since the public key will probably not change
```

Comment:
Since for now, the personal social channels are identified by public keys and not ever-changing PoP tokens (which could be the case later on), we can't create new channels at each Roll Call for each member, because with the way channel creation works (and worked before I changed database implementation), it will fail also if the channel already exists and make things way more complicated to handle.

Reply:
Therefore, I left a FIXME for later on, as we could simply create new channels each time if everybody gets a different PoP token each time

Reply:
I'm still confused a bit, because I thought that at the moment PoP tokens were being used for the social media application, but I don't know what you discussed within the social media meetings 🤔
In any case your latest message makes sense, so go ahead :)